padl/dumptools/var2mod.py

```diff
@@ -596,8 +597,30 @@ def find_globals(node: ast.AST, filter_builtins: bool = True) -> Set[Tuple[str,
     return globals_
 
 
+def _check_and_make_increment(var: ScopedName, scope: Scope, scoped_name: ScopedName):
+    """Check and make increment if the ScopedName after increment exists.
+
+    Increment means the increment of *scoped_name.n*.
+    *scoped_name.n* signifies which one of the usages of a *name* in the code does this
+     *scoped_name* points to.
+    0 being the most recent, and increase integer representing the older usages.
+
+    :param var: Target ScopedName for increment
+    :param scope: Scope
+    :param scoped_name: Source ScopedName for increment
+    :return:
+        ScopedName after increment
+    """
+    try:
+        new_scoped_name = ScopedName(var.name, scope, var.n + scoped_name.n + 1)
+        find_in_scope(new_scoped_name)
+    except NameNotFound:
+        new_scoped_name = ScopedName(var.name, scope, var.n + scoped_name.n)
+    return new_scoped_name
+
+
 def increment_same_name_var(variables: List[ScopedName], scoped_name: ScopedName):
-    """Go through *variables* and increment the the counter for those with the same name as
+    """Go through *variables* and increment the counter for those with the same name as
```

Comment:
Could you add to the end of the docstring `... with the same name as *scoped_name* by incrementing *scoped_name.n*` to make it more clear what we are doing?

Reply:
Never mind this doesn't make sense.